Tasks/Common/AzureRmDeploy-common/azure-arm-rest/azure-arm-app-service-kudu.ts

```diff
@@ -458,9 +458,12 @@ export class Kudu {
         httpRequest.method = 'POST';
         httpRequest.uri = this._client.getRequestUri(`/api/zipdeploy`, queryParameters);
         httpRequest.body = fs.createReadStream(webPackage);
+        let requestOptions = new webClient.WebRequestOptions();
+        requestOptions.retriableStatusCodes = [500, 502, 503, 504];
+        requestOptions.retryIntervalInSeconds = 5;
 
         try {
-            let response = await this._client.beginRequest(httpRequest, null, 'application/octet-stream');
+            let response = await this._client.beginRequest(httpRequest, requestOptions, 'application/octet-stream');
```

Comment:
wouldn't the web client automatically retry here for these codes?

Reply:
yes it will. but it will retry for 408 and 409 as well 
https://github.com/microsoft/azure-pipelines-tasks/blob/bc684d6b2c3ef53937366db18806d9597aa2615f/Tasks/Common/AzureRmDeploy-common/azure-arm-rest/webClient.ts#L50
As per appservice team we should not retry for those codes as it will never turn into succedd

Reply:
can you add this as a comment where you are setting the retriable status codes that this is what the app service team has suggested

Reply:
added comment